operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/KeycloakSpec.java

```diff
@@ -30,6 +30,8 @@ public class KeycloakSpec {
     private int instances = 1;
     @JsonPropertyDescription("Custom Keycloak image to be used.")
     private String image;
+    @JsonPropertyDescription("Secret(s) that might be used when pulling an image from a private container image registry or repository.")
+    private List<LocalObjectReference> imagePullSecrets;
```

Comment:
`LocalObjectReference` only contains, as valid data, a `String`.
I would say that is preferable to opt for the simpler approach of mapping it to a plain `String` to make it easier to handle it with a plain `yaml`.

Reply:
I see really no harm in using `LocalObjectReference` as we are fully aligned with the K8s client and its API this way – no need to do any conversion at the Operator side, and we're providing the same experience to the user as native K8s API.

Reply:
Does the generated yaml object contains `x-kubernetes-preserve-unknown-fields`?

Reply:
I happen to know why are you asking this. And no, `x-kubernetes-preserve-unknown-fields` is not there and therefore the `additionalProperties` of `LocalObjectReference` don't work. But AFAIK this is ok and actually aligned with K8s behaviour at this moment. Just try creating:
```
apiVersion: v1
kind: Pod
...
spec:
  ...
  imagePullSecrets:
  - name: secret
    someAdditionalField: blahblah
```

Reply:
Correct, given `JsonIgnore` everything flows, I simply find it redundant to have an object with a single `String` field in the top-level API:
https://github.com/keycloak/keycloak-k8s-resources/blob/02a2913e2ad9e3454844bd8f6bb0ebfb03765c0e/kubernetes/keycloaks.k8s.keycloak.org-v1.yml#L860-L866

just my 2 cents 🙌

Reply:
Thank you for your input but let's stick to aligning with K8s APIs as we did in the past.

Reply:
I kept the implementation using the `LocalObjectReference` type, as stated by @vmuzikar . 

Thanks for your review, comments and suggestions.